envoy/config/grpc_mux.h

```diff
@@ -21,18 +21,24 @@ using ScopedResume = std::unique_ptr<Cleanup>;
 /**
  * All control plane related stats. @see stats_macros.h
  */
-#define ALL_CONTROL_PLANE_STATS(COUNTER, GAUGE, TEXT_READOUT)                                      \
+#define ALL_CONTROL_PLANE_STATS(COUNTER, GAUGE, TEXT_READOUT, HISTOGRAM)                           \
   COUNTER(rate_limit_enforced)                                                                     \
   GAUGE(connected_state, NeverImport)                                                              \
   GAUGE(pending_requests, Accumulate)                                                              \
+  COUNTER(discovery_requests)                                                                      \
+  HISTOGRAM(discovery_request_resource_count, Unspecified)                                         \
```

Comment:
I think this may be misleading because of the way that the xDS-protocol is built.
The xDS-protocol uses the request to send an ACK/NACK and that will impact the number of resources that are being "sent" in a request. What is desired metric that you wish to capture here?